src/index.js

```diff
@@ -234,6 +217,30 @@ async function createCheck(octokit, context) {
 	};
 }
 
+async function installDependencies(revision) {
+	const cwd = process.cwd();
+	if (await fileExists(path.resolve(cwd, '.pnp.js'))) {
+		return; // No need to install dependencies in projects with PnP enabled.
```

Comment:
This is not correct, it should run an install either way as the `.yarn/unplugged` folder is usually not commited.

You can detect Yarn 2 by checking the `yarn.lock` file for `__metadata:` and running `yarn --immutable` to install

Reply:
Hm, interesting. I guess we can indeed include #41 into this PR to offer generic Yarn 2 support. But the [Zero Install documentation](https://yarnpkg.com/features/zero-installs) suggests people may also check-in the unplugged directories (indeed, we do for our project). Now I could simply detect whether the unplugged directory exists, and if it does we still don't need to install. But Git doesn't maintain empty directories... so if it doesn't exist, does that mean we should install (at least it's the safe thing to do) or doesn't it exist because no packages had postinstall scripts? Would it be too convoluted to check `.gitignore`, and assume we need to do the install if we see `unplugged` is listed there?

Reply:
The easiest and safest way forward is to always trigger the install, if you're using Zero-Installs and commit everything it's basically a noop.

> I guess we can indeed include #41 into this PR to offer generic Yarn 2 support

The existance, or lack thereof, of a `.yarnrc.yml` file isn't enough, the correct way to check is to grep `yarn.lock` for `__metadata:`

Reply:
Do you have a reference to somewhere the `__metadata` is documented? I do believe you, but it feels like relying on an incidental side-effect of the lock file to detect the version.

Reply:
I did find this in the docs: https://yarnpkg.com/getting-started/migration

> The main change is the name of the file. Yarn 1 used `.yarnrc`, but Yarn 2 is moving to a different name: `.yarnrc.yml`. This should make it easier for third-party tools to detect whether a project uses Yarn 1 or Yarn 2, and will allow you to easily set different settings in your home folders when working with a mix of Yarn 1 and Yarn 2 projects.

Reply:
> Do you have a reference to somewhere the __metadata is documented?

I don't, but as one of the maintainers of Yarn I can suggest using this to detect it.

> I did find this in the docs: https://yarnpkg.com/getting-started/migration

That was true at the time it was written, but once we get https://github.com/nodejs/corepack released with node that file is not guaranteed to exist (it isn't either way but oh well).